### PR TITLE
[ci] GitHub Actions: Merge CPU & GPU tasks

### PR DESCRIPTION
**Pull request by feisuzhu**
Issue: #6445

### CODE REVIEW

**Comment by feisuzhu** [comment]
@ailzhang  Please help merge this

**Review by ailzhang — APPROVED** [review]
Thanks!

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -395,7 +306,7 @@ jobs:
              /home/dev/taichi/.github/workflows/scripts/unix_test.sh
         env:
           PY: py38
-          TI_WANTED_ARCHS: 'cuda,vulkan,opengl'
+          TI_WANTED_ARCHS: 'cpu,cuda,vulkan,opengl'
```
Comment (ailzhang) [inline .github/workflows/testing.yml:309]: just to make sure we don't accidentally reduce coverage, shall we keep `cc` in this list as well? 
(which should be removed in a separate PR along with cc backend if we want. )

Comment (feisuzhu) [reply .github/workflows/testing.yml]: Added `cc` to the list

**Review by ailzhang — APPROVED** [review]
Thanks!
